web-client/src/presenter/sequences/signOutIdleSequence.ts

```diff
@@ -12,9 +13,17 @@ export const signOutIdleSequence = [
     // multiple tabs broadcast the idle sign out event.
     no: [],
     yes: [
-      setLogoutTypeAction(LOGOUT_BROADCAST_MESSAGES.idleLogout),
-      signOutSequence,
-      setupCurrentPageAction('IdleLogout'),
+      checkClientNeedsToRefresh,
+      {
+        clientDoesNotNeedToRefresh: [
+          setLogoutTypeAction(BROADCAST_MESSAGES.idleLogout),
+          signOutSequence,
+          setupCurrentPageAction('IdleLogout'),
+        ],
+        // If the client needs to refresh, the sign-out will fail,
+        // and this can lead to inconsistent front-end behavior.
+        clientNeedsToRefresh: [],
+      },
```

Comment:
Although we clear the idle activity interval in IdleActivityMonitor, there is still be a brief period in which this could be called before the interval has been cleared.